Unify button focus styles in login, sign up and onboarding stepper nav

adamwoodnz opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/wp-calypso/issues/99615

## Proposed Changes

Currently the Login and Signup flows have no visible focus styles for most buttons. This PR implements consistent styles across all the buttons in these flows:

* Replaces all buttons with `@wordpress/components` Button components
* Updates the base styles for `@wordpress/components` to reduce duplication of rules like font-size
* Removes many custom styles from buttons which block the base styles from showing

Note that no changes have been made within the onboarding screens other than login and signup; only the nav buttons (Back, Next, Skip) have changed.

### Screenshots

### WordPress.com

| Login | Create account |
|-|-|
| ![calypso localhost_3000_log-in_redirect_to=%2Fsetup%2Fonboarding%2Fdomains signup_url=%2Fsetup%2Fonboarding%2Fuser(2 1366x768)](https://github.com/user-attachments/assets/8d528e4b-46e3-40b7-970f-b722e9ebcedd) | ![calypso localhost_3000_setup_onboarding_start_ref=http%3A%2F%2Fcalypso localhost%3A3000%2F(2 1366x768)](https://github.com/user-attachments/assets/e0c93856-1754-41ae-a197-54fe839971a1) |

| 2FA | Email link | Jetpack |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=%2Fsetup%2Fonboarding%2Fdomains signup_url=%2Fsetup%2Fonboarding%2Fuser email_address=adamwood32%40hotmail com is_signup_existing_account=true(2 1366x768) (1)](https://github.com/user-attachments/assets/841bc3a7-5717-4bf7-a02c-e1f29623f56d) | ![calypso localhost_3000_log-in_redirect_to=%2Fsetup%2Fonboarding%2Fdomains signup_url=%2Fsetup%2Fonboarding%2Fuser email_address=adamwood32%40hotmail com is_signup_existing_account=true(2 1366x768)](https://github.com/user-attachments/assets/c1f97858-5e77-4598-87e9-ee4a8140a29b) | ![Screenshot 2025-02-25 at 5 01 49 PM](https://github.com/user-attachments/assets/a678ff8a-04c0-4d2c-b143-790af5c5eaf0) |

| User onboarding | Current user |
|-|-|
| ![calypso localhost_3000_setup_onboarding_user(2 1366x768)](https://github.com/user-attachments/assets/affbbfdb-0f95-4dab-9216-fc03cd4e8eb4) | ![calypso localhost_3000_log-in_link(2 1366x768)](https://github.com/user-attachments/assets/3048e84b-e044-4afd-bfc3-1ad29d2a9ed8) |

#### Crowdsignal

| Login | Signup | Continue as user |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D4d11c652d720c922e2eab1ea0c27e15b6bbe585a1f2610a20176a84394cca2ef%26redire](https://github.com/user-attachments/assets/b0ba7ed0-0172-4b90-b7bc-ec80ac9ac816) | ![calypso localhost_3000_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D4d11c652d720c922e2eab1ea0](https://github.com/user-attachments/assets/b681eb54-1a00-405d-bfdc-dd0d510e2d68) | ![calypso localhost_3000_start_crowdsignal_oauth2-name_oauth2_client_id=978 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D4d11c652d720c922e2eab (1)](https://github.com/user-attachments/assets/221ae0ec-619e-4c56-bf61-e1cdac9a074a) |

#### Woo

| Login | Signup | Continue as user |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Db30d4e6a81f72b803c873f517f4189601c46a1bc8b9bfdb9ed27cf3d17ad5601%26redirect_uri%3D](https://github.com/user-attachments/assets/d4f3d5d1-0ba7-43f5-9b1f-f502d75942cd) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=50916 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Db30d4e6a81f72b803c873f517f4189601c4 (1)](https://github.com/user-attachments/assets/c8470cf2-3fdb-46da-b3d5-91d9f6c049ee) | ![calypso localhost_3000_log-in_webauthn_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Db30d4e6a81f72b803c873f517f4189601c46a1bc8b9bfdb9ed27cf3d17ad5601%26redire](https://github.com/user-attachments/assets/22265a58-868c-43d2-9a4a-f0c3f6866bcc) |

#### Jetpack

| Login | Signup |
|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud jetpack com%252Fconnect%252Foauth%252Ftoken%253Fnex](https://github.com/user-attachments/assets/ef873843-782e-49b0-89b0-d83b362c8ae9) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=69040 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud jetpack](https://github.com/user-attachments/assets/00df2406-f69d-4804-9e90-eade74a5896b) |

#### Blaze Pro

| Login | Signup | Continue as user | Reset password |
|-|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb190dc03654cef0465786da0%26redirect_uri%3 (1)](https://github.com/user-attachments/assets/9e9ad632-204a-4909-8972-a253ac4e5eed) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=99370 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb1](https://github.com/user-attachments/assets/1ab6e2fc-0f17-482c-9b49-b0b25902e4a9) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb190dc03654cef0465786da0%26redirect_uri%3Dhtt](https://github.com/user-attachments/assets/9d9ac7bb-1b47-4c93-a532-4d327bbbd82e) | ![calypso localhost_3000_log-in_lostpassword_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb190dc03654cef0465786da0%26redir](https://github.com/user-attachments/assets/21afa27d-4947-45d3-97cf-4d837bed24c5) |

#### A8C for Agencies

| Login | Signup |
|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95931%26redirect_uri%3Dhttps%253A%252F%252Fagencies automattic com%252Fconnect%252Foauth%252Ftoken%2](https://github.com/user-attachments/assets/ddf9d3a6-6514-4841-855d-b890043609bd) | ![calypso localhost_3000_start_wpcc_oauth2-user_oauth2_client_id=95931 oauth2_redirect=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95931%26redirect_uri%3Dhttps%253A%252F%252Fagencies autom](https://github.com/user-attachments/assets/f4af0b3b-e3dd-463d-ad88-779c829650be) |

#### Akismet

| Login | Signup | Continue as user |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fr-login wordpress com%2Fremote-login php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet com%252Faccount%252F signup_url=%2Fstart%2Faccount%2Fuser-social%3Fredirect_to%3Dhttps (1)](https://github.com/user-attachments/assets/814a5c42-ecbb-4c48-a1ff-709cdd376e5e) | ![calypso localhost_3000_start_account_user-social_redirect_to=https%3A%2F%2Fr-login wordpress com%2Fremote-login php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet com%252Faccount%252F(Desktop)](https://github.com/user-attachments/assets/699a3a6f-7dfd-484a-a9e4-a20e380d84e0) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fr-login wordpress com%2Fremote-login php%3Faction%3Dlink%26back%3Dhttps%253A%252F%252Fakismet com%252Faccount%252F signup_url=%2Fstart%2Faccount%2Fuser-social%3Fredirect_to%3Dhttps%253](https://github.com/user-attachments/assets/107cb3f4-c919-46c3-808f-9c19763385db) |

#### Gravatar

| Login | By username | Signup |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D4a5153e30a28e2ef24ca7e599abd80f3969340cd1f88327c1f5d2fbc88593fa1%26redir](https://github.com/user-attachments/assets/3a05d0e3-e985-4567-92dc-a9ec1f28d4cd) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D4a5153e30a28e2ef24ca7e599abd80f3969340cd1f88327c1f5d2fbc88593fa1%26r (2)](https://github.com/user-attachments/assets/2643a2b6-6335-45a3-9fcd-3275af872ead) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D4a5153e30a28e2ef24ca7e599abd80f3969340cd1f88327c1f5d2fbc88593fa1%26r (1)](https://github.com/user-attachments/assets/604f01cf-c9f4-47f6-8d35-52149c5266bf) |

#### WP Job Manager

| Login | By email | Signup |
|-|-|-|
| ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26redi](https://github.com/user-attachments/assets/e76c31d6-1897-45a2-ae23-4600cb6cc9a8) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26 (2)](https://github.com/user-attachments/assets/2efd490e-0701-424b-acb5-348c52df3526) | ![calypso localhost_3000_log-in_redirect_to=https%3A%2F%2Fpublic-api wordpress com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26 (1)](https://github.com/user-attachments/assets/d80dbd02-944f-4f74-b1a9-d716a256a8aa) |


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Visible focus styles are important for accessibility
* Consistent styles and behaviour across interactive elements leads to a more polished, high quality experience

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Ensure you are logged out
* Visit http://calypso.localhost:3000/start/account/user-social, http://calypso.localhost:3000/log-in, http://calypso.localhost:3000/setup/onboarding/user
* Use the tab key to move focus through the page
* Navigate into other screens in the flow (eg. 2FA after entering password) and continue tabbing to check focus styles
* Observe that all the buttons have matching visible focus styles

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

adamwoodnz committed:
Remove custom focus styles from social buttons

adamwoodnz committed:
Add accessible focus handler to login block

adamwoodnz committed:
Add test for calling accessibleFocus on mount

adamwoodnz committed:
Remove box shadow from social buttons in Jetpack and A4A login form

adamwoodnz committed:
Move accessibleFocus call to LayoutLoggedOut to cover all login and account creation screens

adamwoodnz committed:
Move test

adamwoodnz committed:
Split test file into separate ssr and client contexts

adamwoodnz commented:
> [!IMPORTANT] 
> This is an AI-generated review by https://github.com/Automattic/pr-review-agent/ 

Hey there! I've taken a look at your PR for adding focus styles to the social login buttons. The changes look good overall, and I appreciate the attention to accessibility. I have a couple of suggestions that might help improve the implementation:

1. While removing the box-shadow styles that were blocking the base focus styles is a good start, I notice you're relying on the base focus styles. Given that this is a key user interaction point, we might want to consider defining explicit focus styles for these social buttons. This would give us more control over the visual appearance and ensure consistency across different browsers. You could add something like:

```scss
.auth-form__social-buttons {
  button {
    &:focus-visible {
      outline: 2px solid var(--color-primary);
      outline-offset: 2px;
    }
  }
}
```

2. The accessible focus handler is a great addition for keyboard users. However, I noticed in the test files that there isn't specific coverage for testing the focus styles. Consider adding a test case in `client/blocks/login/test/login-form.jsx` that verifies the accessible focus class is properly applied when using keyboard navigation.

The changes look solid otherwise! The screenshots show clear focus indicators, and I like that you're planning ahead for the Blueberry color scheme update. Let me know if you'd like me to clarify anything about these suggestions.

adamwoodnz committed:
Revert "Remove box shadow from social buttons in Jetpack and A4A login form"

This reverts commit d8714bbe64c302904b30b764121da7e212ad7a1e.

jasmussen reviewed (approved):
Thank you, this is an outstanding step forward, and ready to land.

![state](https://github.com/user-attachments/assets/4715d531-cb65-40dc-9e9a-b8de796d43a6)

Note in the GIF above, that there are additional issues, I'm wondering if we should track this separately, it seems such a deep issue. It might even be worth discussing whether to remove the accessible-focus utility; if need be it can even be replaced with situational focus-visible, which seems a better way to do it. 

The issues, blink and you'll miss it:

1. The primary button does not have a focus style.
2. All the "Continue with" buttons are _Calypso_ components, whereas the email and jetpack buttons are Gutenberg components. The latter come with a subtle fading effect to the focus style, the former do not.
3. Minor: the link does not have a focus style and falls back to the dotted browser default. It would be a better look with a single unified focus style across everything.

The real solution is to replace the Calypso components with Gutenberg components, though that's obviously a larger endeavour, and not in any way a blocker for this good step forward, or even urgent in their own right. It's just a slew of small papercuts that add up, and I'm mainly wondering your instincts on how to best track subsequent work here. Happy to open issues.

adamwoodnz commented:
> The real solution is to replace the Calypso components with Gutenberg components, though that's obviously a larger endeavour, and not in any way a blocker for this good step forward, or even urgent in their own right. It's just a slew of small papercuts that add up, and I'm mainly wondering your instincts on how to best track subsequent work here.

@jasmussen yeah I have made most of the same observations. My intention with this PR is to remove all the things which stop focus styles being displayed, so at least we meet accessibility standards. In https://github.com/Automattic/wp-calypso/pull/99915 I have started updating the existing focus styles for all the button variants (including primary) to match Gutenberg, but I hadn't considered actually replacing them with Gutenberg components. I think this sounds great (and feels on point with untangling) if we can do it incrementally, and I'll spike this approach with these screens to see how it shakes out. My concern is whether the existing accessible-focus styles would mess with the Gutenberg styles. If we've scoped both sets well it might not be an issue.

adamwoodnz committed:
Migrate all login and create account buttons to WordPress components

adamwoodnz committed:
Revert adding accessible focus to logged out layout

adamwoodnz commented:
> I'll spike this approach with these screens to see how it shakes out. My concern is whether the existing accessible-focus styles would mess with the Gutenberg styles. If we've scoped both sets well it might not be an issue.

Ok, I've reverted the accessible focus addition and replaced all the buttons with `@wordpress/components` buttons. Works pretty well! Will need much more thorough testing now to check that all the buttons still work, but design wise we're in a good place I think. Most overrides are gone, there are a few customizations to the social button secondary buttons remaining.

![calypso localhost_3000_log-in_redirect_to=%2Fsites signup_url=%2Fstart%2Faccount%2Fuser-social%3Fredirect_to%3D%252Fsites(2 1366x768)](https://github.com/user-attachments/assets/633f88f1-684a-4589-8bc7-6f15c18f6e4c)

adamwoodnz committed:
Reduce scope of login button change

jasmussen commented:
Really nice work! Tested, this still works well. The focus styles were already inconsistent, so the fact that there's a single set of components that are now actually consistent, that's a good start. 

![state](https://github.com/user-attachments/assets/5a33a4f6-cb47-41c8-b0b6-3e27700cd7fa)


> I have started updating the existing focus styles for all the button variants (including primary) to match Gutenberg, but I hadn't considered actually replacing them with Gutenberg components. I think this sounds great (and feels on point with untangling) if we can do it incrementally, and I'll spike this approach with these screens to see how it shakes out. My concern is whether the existing accessible-focus styles would mess with the Gutenberg styles. If we've scoped both sets well it might not be an issue.

Right, we need to carefully balance the effort here, I trust you to be mindful of your time and any competing priorities. This is papercut/product quality work, which is important, but nevertheless secondary to anything else that might be on your table.

Because it's that category of work, it also makes sense to mainly do this right, just as you did here. The mix of component types here seems an oversight or rushed part of the puzzle, and the amount of deleted code in the update reflects that it's on a good path. Nice one!

adamwoodnz committed:
Use WordPress buttons for the remaining login forms

adamwoodnz commented on client/login/magic-login/style.scss:
```diff
@@ -153,23 +156,10 @@
-			color: var(--color-primary);
-		}
-		&:last-of-type {
-			border-bottom: none;
-		}
 
```
I can't find an instance where there are 2 anchors in a magic login footer, so these border styles seem safe to remove.

adamwoodnz commented on client/login/magic-login/style.scss:
```diff
@@ -153,23 +156,10 @@
 	margin: 4px 0 48px 0;
+	text-align: center;
 
 	a {
```
This has become a link Button

adamwoodnz committed:
Customize wordpress button height and remove more overrides

adamwoodnz committed:
Replace button on sign up form in onboarding

adamwoodnz commented:
I've moved this back to in progress while I make the whole flow consistent

adamwoodnz committed:
Remove unnecessary border radius

adamwoodnz commented:
![Screenshot 2025-02-24 at 3 35 38 PM](https://github.com/user-attachments/assets/bb7484fa-b545-4f6d-83c4-6f2fd2dd95a1)

@jasmussen in the above screenshot of the onboarding sign up form you can see that all the link buttons have different focus styles. Two of them are `@wordpress/components` `isLink` buttons with custom styles to make them dark grey and semi bold (see inspector), one is just a styled link. I've replaced a couple of these in other screens so they are now the default blueberry color, eg. 'Lost your password?' below:

![calypso localhost_3000_log-in_redirect_to=%2Fsetup%2Fonboarding%2Fdomains signup_url=%2Fsetup%2Fonboarding%2Fuser(2 1366x768)](https://github.com/user-attachments/assets/44cabe53-e292-4518-8ac4-8c5c6b617b14)

Having a consistent focus ring is a given and I'll remove any styles which block the Gutenberg style for that, but what do you think about the unfocused state; should we remove the overrides to make them the default Blueberry color across the whole flow? Do we need to consult anyone about this?

adamwoodnz committed:
Remove custom focus styles from signup forms back button

jasmussen commented:
> Having a consistent focus ring is a given and I'll remove any styles which block the Gutenberg style for that, but what do you think about the unfocused state; should we remove the overrides to make them the default Blueberry color across the whole flow? Do we need to consult anyone about this?

This is blame I'm willing to wear proudly, so we don't need to consult anyone, and point them to me if anything. Feel free to remove custom overrides for focus styles, wherever you find them, so long as what we replace them with is _a coherent default_, and that coherent default is ideally:

* Blueberry
* 2px radius
* 1.5px thick

Took the work for a pass, by the way, the approval still stands:

![fix](https://github.com/user-attachments/assets/36758042-44b6-4a66-aff7-8bfdee65ddc2)

Here's "before":

![before](https://github.com/user-attachments/assets/4e99af7e-b3f6-4ff3-a031-c56ef12a11e1)

Note that the gap between the email input and the primary button is now smaller. I don't think that's a blocker, this particular gap is also something that should be inherited by default componentry, so just flagging it in case there's something missing.

adamwoodnz commented:
Tasks remaining:

- [ ] Check for login and signup flow e2e tests
- [ ] Remove any remaining custom focus styles from buttons
- [ ] Remove any custom styling from link buttons below forms
- [ ] Smoke check output HTML markup of buttons is the same as prod
- [ ] Notify slack channels that this is shipping
- [ ] Ship
- [ ] Monitor

adamwoodnz committed:
Use a WordPress components button for stepper navigation links

adamwoodnz committed:
Make the Jetpack qr code screen back link consistent

adamwoodnz committed:
Remove custom styles from onboarding step sign up form back button

adamwoodnz committed:
Make sign up form back button consistent

adamwoodnz committed:
Remove custom focus styles from stepper nav link buttons

adamwoodnz committed:
Reverse import naming now that A8C buttons are deprecated

adamwoodnz committed:
Use WordPress button for continue as logged in user screen

adamwoodnz committed:
Use WordPress button for sign up screen log in link

adamwoodnz committed:
Reduce scope of changes to step navigation link

adamwoodnz commented on packages/onboarding/src/step-navigation-link/index.tsx:
```diff
@@ -24,7 +24,6 @@ const StepNavigationLink: React.FC< Props > = ( {
 	hasBackIcon,
 	hasForwardIcon,
 	primary,
-	borderless = true,
```
This default is now handled by the `getVariant` function; it returns the WordPress button equivalent `variant: 'tertiary'`

adamwoodnz committed:
Reverse import naming now that A8C button is deprecated

adamwoodnz committed:
Revert reduction in height of social buttons

adamwoodnz committed:
Adjust default border radius and font size of WordPress button

adamwoodnz committed:
Fix regression on step navigation next button

adamwoodnz committed:
Revert adding margin to continue as user screen

adamwoodnz committed:
Fix alignment of signup email form back button on Safari

adamwoodnz committed:
Revert removing styles for other continue as user buttons

adamwoodnz commented on client/blocks/login/continue-as-user.jsx:
```diff
@@ -98,14 +99,14 @@ export default function ContinueAsUser( {
 					</div>
 				</div>
-				<Button
+				<A8CButton
```
I'm not familiar with (and don't know how to test) these other product screens so for now I've left them and focused on dotcom

adamwoodnz commented on client/blocks/login/continue-as-user.jsx:
```diff
@@ -1,4 +1,5 @@
-import { Button } from '@automattic/components';
+import { Button as A8CButton } from '@automattic/components';
```
This button has been marked as [deprecated](https://github.com/Automattic/wp-calypso/blob/fffc4d2b1a4f4e0f7c6574419234b1e450838689/packages/components/src/button/index.tsx#L108) so I've chosen to make `Button` the WordPress components button, as that's what we should be using from now on.

adamwoodnz commented on client/blocks/login/login-form.jsx:
```diff
@@ -1089,13 +1094,14 @@ export class LoginForm extends Component {
 							{ ! isBlazePro && <p className="login__form-terms">{ socialToS }</p> }
 							{ shouldRenderForgotPasswordLink && this.renderLostPasswordLink() }
 							<div className="login__form-action">
-								<FormsButton
```
[FormsButton](https://github.com/Automattic/wp-calypso/blob/fffc4d2b1a4f4e0f7c6574419234b1e450838689/client/components/forms/form-button/index.jsx#L9) is basically an A8C submit Button with some default text, which won't be used here because children are being supplied.

adamwoodnz commented on client/blocks/signup-form/style.scss:
```diff
@@ -295,41 +267,6 @@ body.is-section-accept-invite {
 				flex-direction: column;
 				width: 100%;
-
-				.social-buttons__button {
```
These styles are now centralized in `client/components/social-buttons/style.scss`

adamwoodnz commented on packages/onboarding/src/step-container/index.tsx:
```diff
@@ -140,7 +140,6 @@ const StepContainer: React.FC< Props > = ( {
 					cssClass={ clsx( 'step-container__navigation-link', 'has-underline', {
 						'has-skip-heading': skipHeadingText,
 					} ) }
-					borderless
```
This is the default, so does not need to be passed

adamwoodnz commented on packages/onboarding/src/step-container/index.tsx:
```diff
@@ -158,7 +157,6 @@ const StepContainer: React.FC< Props > = ( {
 				handleClick={ goNext }
 				label={ nextLabelText || translate( 'Continue' ) }
 				cssClass="step-container__navigation-link"
-				borderless={ false }
```
This has `primary: true` so isn't necessary to be passed

adamwoodnz committed:
Center WordPress Button content by default

adamwoodnz committed:
Replace buttons in Crowdsignal screens

adamwoodnz commented on client/blocks/login/style.scss:
```diff
@@ -236,6 +236,7 @@ $breakpoint-mobile: 782px; //Mobile size.
 .login__form-action {
 	button {
 		width: 100%;
+		height: 44px;
```
Login form inputs and submit buttons are taller across the board (44px vs 40px). This might be temporary if we decide to standardize them.

adamwoodnz commented on client/blocks/signup-form/crowdsignal.scss:
```diff
@@ -174,9 +171,20 @@ body.is-section-signup .layout .signup.is-crowdsignal .step-wrapper {
 	display: flex;
+	height: 44px;
 	margin-bottom: 40px;
+	padding-inline: 16px;
+
+	> svg {
+		margin-right: auto;
+	}
+
+	> span {
+		margin-left: -20px;
+		margin-right: auto;
+	}
```
Has the WordPress logo on the left, so needs to match the layout of the social buttons below

adamwoodnz committed:
Replace buttons in Woo screens

adamwoodnz committed:
Fix syntax error

chihsuan reviewed (commented):
Thank you @adamwoodnz! I tested the woo login and signup flows and can confirm that the focus styles are much better than before. 👍   I think using the Woo's purple color for the focus styles would be better, but we can do that later.

I've observed that the border and icon position of the social buttons have been altered. It would be preferable to retain the original design. Could you ensure that the original design is preserved? Thank you!


Before:
<img width="281" alt="Screenshot 2025-02-27 at 11 31 19" src="https://github.com/user-attachments/assets/8e74fdc3-c481-4155-bcf4-d74ed15d962f" />

After:
<img width="282" alt="Screenshot 2025-02-27 at 11 31 14" src="https://github.com/user-attachments/assets/5fccbf46-52e3-4599-85d8-f4cb02b5a404" />

Figma:
tp2u3Nk9arcpeNMZcylNG4-fi-271_8254

adamwoodnz committed:
Replace Jetpack Cloud buttons

chihsuan commented:
> Edit: Correct me if I'm missing context. In case this very same componentry and flow is used in a non wp.com context, it can make sense

Thanks for the feedback! 🙏  @jasmussen I agree all focus styles should be the same in the wp.com context. 🙂 

I was referring to the account creation flow on woocommerce.com. I believe we should use the same focus style as woocommerce.com since users are signing up or logging in on woocommerce.com. 

> though in this case I would expect the modern focus style to be refrencing an "accent" CSS variable that hopefully can be supplied from the Woo side of things.

Make sense!

adamwoodnz committed:
Restore woo social button layout

adamwoodnz committed:
Use components vars to change button colors

adamwoodnz committed:
Use WordPress buttons for Blaze Pro

adamwoodnz committed:
Use vars to theme Woo buttons

adamwoodnz committed:
Make Woo login button full width

adamwoodnz committed:
Remove unnecessary justification

adamwoodnz commented:
> So if you change `--wp-components-color-accent` to Woo Purple, you should be set. With that, however, I'm also suggesting that the right fix is to use WordPress components in as many places as possible, so we can have similar benefits on other contexts.

I've changed my approach to leverage these variables, so the individual products use themed accent colors for buttons now. See the updated screenshots in the description and [ca048c8](https://github.com/Automattic/wp-calypso/pull/99693/commits/ca048c886ba4a11cc1bbc04e86160dfe788a882d)

adamwoodnz committed:
Stop Blaze Pro styles leaking into other products

adamwoodnz committed:
Reduce duplication of login footer link styles

adamwoodnz committed:
Fix Woo continue as user button text color

adamwoodnz committed:
Update Jetpack signup snapshot

adamwoodnz committed:
Update Jetpack signup snapshot

adamwoodnz commented on client/jetpack-connect/test/__snapshots__/signup.js.snap:
```diff
@@ -212,8 +212,7 @@ exports[`JetpackSignup should render 1`] = `
               </p>
               <button
-                class="button signup-form__submit form-button is-primary"
-                locale="en"
```
I cannot find any code which depends on this non-standard attribute. I'd expect if it was used it would be `data-locale`. [It's passed to all `FormButton` components via the `localize` higher order component](https://github.com/Automattic/wp-calypso/blob/b159deb4ada78061bb88437f5d7e50e460251c4a/client/components/forms/form-button/index.jsx#L38), and is removed now that we're using a WordPress Button directly.

chihsuan reviewed (approved):
Awesome! Tested well on the Woo login and signup flows. 👍 Thank you! @adamwoodnz

adamwoodnz committed:
Use Akismet colors in buttons

adamwoodnz committed:
Restore width of continue as user primary button

adamwoodnz committed:
Make Akismet redirection detection secure

adamwoodnz committed:
Simplify Akismet styles

adamwoodnz committed:
Add tests for isAkismetRedirect

adamwoodnz committed:
Style Gravatar buttons

adamwoodnz committed:
Add Gravatar and WP Job Manager accent colors to default color scheme

adamwoodnz committed:
Style WP Job Manager buttons

adamwoodnz committed:
Expand isAkismetRedirect test to handle empty string input

adamwoodnz committed:
Replace reset password form button

adamwoodnz committed:
Unify login links layout

adamwoodnz committed:
Style P2 buttons

adamwoodnz commented:
I'm no longer comfortable with the size and scope of this PR so I'm going to investigate ways of breaking it into smaller chunks. At this stage I'm thinking something like:

1. Primary buttons on login
2. Primary buttons on signup
3. Social buttons (login and signup)
4. Footer links
...

This will make it easier to test, even if it does affect 10 products each time (.com, Crowdsignal, Woo, P2, VIP, Jetpack Cloud, Akismet, A4A, WP Job Manager, Gravatar, maybe more?).
